spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java

```diff
@@ -63,29 +70,21 @@
 import org.apache.spark.sql.SaveMode;
 import org.apache.spark.sql.SparkSession;
 import org.apache.spark.sql.catalyst.InternalRow;
-import org.assertj.core.api.Assertions;
-import org.junit.AfterClass;
-import org.junit.Assert;
-import org.junit.Assume;
-import org.junit.BeforeClass;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-
-@RunWith(Parameterized.class)
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(ParameterizedTestExtension.class)
 public class TestDataFrameWrites extends AvroDataTest {
   private static final Configuration CONF = new Configuration();
 
-  private final String format;
-
-  @Parameterized.Parameters(name = "format = {0}")
-  public static Object[] parameters() {
-    return new Object[] {"parquet", "avro", "orc"};
+  @Parameters(name = "format = {0}")
+  public static Collection<String> parameters() {
+    return Arrays.asList("parquet", "avro", "orc");
   }
 
-  public TestDataFrameWrites(String format) {
-    this.format = format;
-  }
+  @Parameter private String format = "parquet";
```

Comment:
it is initialised automatically for the @TestTemplates tests in TestDataFrameWrites, but not for the @Tests it inherits from class `AvroDataTest`. 
You don't see this problem with other classes that inherit from `AvroDataTest` because their @Parameter variable is a boolean, which is False by default. 
@Parameter `format` being a string needs a default non-null value, otherwise it throws errors

Reply:
hm I see, I didn't realize that

Reply:
@chinmay-bhat I pushed a commit to fix this and some other places that were wrongly using parameterized tests